lib/common/socket.c

```diff
@@ -1589,15 +1615,90 @@ static void on_handshake_fail_complete(h2o_socket_t *sock, const char *err)
 
 static void proceed_handshake(h2o_socket_t *sock, const char *err);
 
+#if H2O_CAN_ASYNC_SSL
+
+static void on_async_proceed_handshake(h2o_socket_t *async_sock, const char *err)
+{
+    if (err != NULL)
+        h2o_fatal("error on internal notification fd:%s", err);
+
+    /* Do we need to handle spurious events for eventfds / pipes used for intra-process communication? If so, we have to read
```

Comment:
By spurious events do you mean epoll reporting read ready twice for ex? The intention is that this callback is only called once for the `async_sock` (`h2o_socket_read_stop` called below)

Reply:
I mean read-ready being reported where in fact nothing can be read. Quoting from [man 2 select](https://linux.die.net/man/2/select):
> Under Linux, select() may report a socket file descriptor as "ready for reading", while nevertheless a subsequent read blocks. This could for example happen when data has arrived but upon examination has wrong checksum and is discarded. There may be other circumstances in which a file descriptor is spuriously reported as ready. Thus it may be safer to use O_NONBLOCK on sockets that should not block. 

I want to assume that such thing would never happen on a pipe / UNIX socket but I'm not entirely certain.